zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -492,7 +492,7 @@ protected String filePathGenerator(String datasetId, Date publishDate, String Da
         String finalPath;
 
         if (nonTsDatasetWhitelistSet.contains(baseFilename)) {
-            finalPath = "ts-datasets/" + "other" + "/" + formattedDate;
+            finalPath = "ts-datasets/" + "other" + "/" + formattedDate + "/" +(!correctDatasetVersion.equals("")? incrementDatasetVersionByOne(correctDatasetVersion) : DatasetVersion);
         } else {
             finalPath = "ts-datasets/" + baseFilename + "/" + (!correctDatasetVersion.equals("")? incrementDatasetVersionByOne(correctDatasetVersion) : DatasetVersion);
```

Comment:
You're duplicating that inline if statement here. Is there a *dry*er way of doing this?

Reply:
Refactored the function and removed duplication using another functionality to get the correct version